### ISSUE TITLE
\Phalcon\Security\Random:base64Safe() function has problem

### ISSUE THREAD

**Issue by AnonymousLab** [phalcon/cphalcon#12141]
Class Phalcon\Security\Random

```php
	public function base64(int len = null) -> string
	{
		return base64_encode(this->bytes(len));
	}

	public function base64Safe(int len = null, boolean padding = false) -> string
	{
		var s;

		let s = preg_replace("#[^a-z0-9_=-]+#i", "", this->base64(len));

		if !padding {
			return trim(s, "=");
		}

		return s;
	}
```

please look at this ↓↓↓

```php
let s = preg_replace("#[^a-z0-9_=-]+#i", "", this->base64(len));
```

this->base64(len) is not include -, _ characters.

so base64Safe() is not include _, - characters.

i think ...

this->base64(len) -> change \, + to  _, -  

after 

preg_replace("#[^a-z0-9_=-]+#i", "", .....

thank you for reading.

**Comment by Green-Cat** [comment]
@sergeyklay

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
